download/download.go

```diff
@@ -79,51 +80,49 @@ func (d *Downloader) getSize(ch chan<- error, url string) {
 
 	r, err := d.client.Head(url)
 	if err != nil {
-		ch <- err
+		ch <- fmt.Errorf("Error sending a HTTP HEAD request to %s: %s", url, err)
 		return
 	}
-	for _, k := range []string{"Content-Length", "content-length"} {
-		var s int
-		s, err = strconv.Atoi(r.Header.Get(k))
-		if err == nil {
-			d.totalSize += int64(s)
-			ch <- nil
-			return
-		}
+	defer r.Body.Close()
+
+	if r.ContentLength == 0 {
+		ch <- fmt.Errorf("Could not get size for %s", url)
+		return
 	}
-	err = fmt.Errorf("Could not get size for %s", url)
-	ch <- err
+
+	d.totalSize += r.ContentLength
```

Comment:
Just noticed that you update this variable from various threads, which leads to a race condition.

Two simple ways to deal with it:
- [sync.Mutex](https://pkg.go.dev/sync#Mutex)
- use the [atomic.AddInt64](https://pkg.go.dev/sync/atomic#AddInt64) 

Even though those work, as you already use a channel, I would suggest a third direction:

Make `ch chan<- error` a chan of a struct, which holds a result or error (example [here](https://stackoverflow.com/a/25142269)).

Then you don't update `d.totalSize` from within the concurrent function, but instead from `getTotalSize`. The code from lines 105 to 107 will change slightly.